### PR TITLE
Postgres: Support request cancellation properly (Uses new backendSrv.fetch Observable request API)

### PR DESCRIPTION
**Pull request by hugohaggmark**
What this PR does / why we need it:
This PR replaces the usage of datasourceRequest in the Postgres data source with fetch and updates tests accordingly.

Which issue(s) this PR fixes:
Fixes #

Relates #27222

Special notes for your reviewer:
The tests where missing test cases for the `query` method so I tried to run some [marble tests](https://rxjs-dev.firebaseapp.com/guide/testing/marble-testing) in this PR to test Observables. Let me know what you think about that :)

### CODE REVIEW

**Review by marefr — APPROVED** [review]
Tested it quickly and seems to work as before :+1: Someone else have to review the marble tests :)

**Comment by hugohaggmark** [comment]
Thank you @marefr 🌟

File: public/app/plugins/datasource/postgres/specs/datasource.test.ts [hunk 1]
```diff
@@ -1,85 +1,282 @@
...
+import { TestScheduler } from 'rxjs/testing';
+import { FetchResponse } from '@grafana/runtime';
 import { dateTime, toUtc } from '@grafana/data';
+
+import { PostgresDatasource } from '../datasource';
 import { backendSrv } from 'app/core/services/backend_srv'; // will use the version in __mocks__
 import { TemplateSrv } from 'app/features/templating/template_srv';
 import { initialCustomVariableModelState } from '../../../../features/variables/custom/reducer';
+import { TimeSrv } from '../../../../features/dashboard/services/TimeSrv';
+import { describe } from '../../../../../test/lib/common';
```
Comment (dprokop) [inline public/app/plugins/datasource/postgres/specs/datasource.test.ts:11]: Think you don't need this since describe is global?

Comment (hugohaggmark) [reply public/app/plugins/datasource/postgres/specs/datasource.test.ts:11]: hmm, WebStorm is not behaving nicely. I'll fix

**Comment by dprokop** [comment]
@hugohaggmark this looks great. I have a hard time looking at the marble tests at the beginning, but after reading [this](https://medium.com/@bencabanes/marble-testing-observable-introduction-1f5ad39231c) first and after that the [official docs](https://rxjs-dev.firebaseapp.com/guide/testing/marble-testing#api) I actually find this a really interesting and readable approch. 

It would be great tho if we got some more input on this approach from the team (adding some more people as reviewers) - and maybe this is a candidate to discuss during front-end weekly?

**Comment by hugohaggmark** [comment]
I have received little feedback against the marble diagrams so I'm going to merge this.

**Pull request merged**